### PR TITLE
Implement the Arbitrary fuzzing trait for contract value-types

### PR DESCRIPTION
**Pull request by brson**
### What

Implement the Arbitrary trait for Symbol, BitSet, Static, and Status.

### Why

This is a prerequisite for more complete fuzzing support in soroban-sdk: https://github.com/stellar/rs-soroban-sdk/pull/878
This patch is only needed if that patch is to be accepted.

This patch also tweaks and reexports the `call_with_suppressed_panic_hook` function,
which is used by the sdk patch to provide users a helper function for catching panics in fuzz tests.

A more complete description is in the above link.

### Known limitations

See sdk PR.

### CODE REVIEW

**Comment by brson** [comment]
Rebased.

**Comment by dmkozh** [comment]
Just a heads up: we are in the middle of a significant env/sdk rewrite that is definitely going to affect this and SDK PRs to some degree. Due to that I would be inclined to wait until the rewrite is landed (should happen in a few days). Thanks for the work and sorry for slow review.

**Comment by brson** [comment]
Rebased.

File: soroban-env-common/src/arbitrary.rs
Comment (dmkozh) [inline soroban-env-common/src/arbitrary.rs:14]: What is the use case for this now? FWIW the SDK doesn't expose `SymbolSmall` explicitly and instead provides a wrapper `Symbol` type that can be either small or env-based value. So maybe it would make more sense to implement this together with SDK fuzzing support (I suppose it would be similar to fuzzing any env type, like bytes or string).

File: soroban-env-common/src/arbitrary.rs
Comment (dmkozh) [inline soroban-env-common/src/arbitrary.rs:11]: There is no name collision here with some other `Vec`, is there? Up to you though.

File: soroban-env-common/src/arbitrary.rs
Comment (brson) [reply soroban-env-common/src/arbitrary.rs:14]: If the SDK doesn't expose this, then it probably isn't needed. I just rebased this week and adapted this code from the prior code for Symbol. I need to go back and reevaluate which types need arbitrary impls.

**Comment by brson** [comment]
Is the `Status` types still exposed by the sdk? If not this patch is no longer needed at all.

File: soroban-env-common/src/arbitrary.rs
Comment (brson) [reply soroban-env-common/src/arbitrary.rs]: I've removed it.

**Comment by brson** [comment]
I removed the implementation of Arbitrary for SymbolSmall. The only Arbitrary implementation left now in this crate is for Status. I'm not clear on whether contract authors can still access this type or not. It may be that none of the non-object contract types need to implement Arbitrary.

**Comment by dmkozh** [comment]
> Is the Status types still exposed by the sdk?

Yeah, it's still exposed, so probably we still need this.

> I removed the implementation of Arbitrary for SymbolSmall. The only Arbitrary implementation left now in this crate is for Status. I'm not clear on whether contract authors can still access this type or not. It may be that none of the non-object contract types need to implement Arbitrary.

`SymbolSmall` is accessible via `soroban-env-common`, but it isn't too useful as SDK exposes a higher level `Symbol` type that can be either small or object-based symbol, so I think it would make more sense to implement Arbitrary for the SDK `Symbol` instead.

**Review by dmkozh — APPROVED** [review]

**Comment by dmkozh** [comment]
Let's wait a bit with merging this to get it into the next release (the current release happens right now and there is also the SDK change which we won't have time to merge into it)

**Pull request merged**
